src/java/org/apache/cassandra/config/EncryptionOptions.java

```diff
@@ -114,19 +165,61 @@ public EncryptionOptions(EncryptionOptions options)
      * @return if the channel should be encrypted
      */
     public boolean isEnabled() {
-        return this.enabled;
+        ensureConfigApplied();
+        return isEnabled;
     }
 
     /**
      * Sets if encryption should be enabled for this channel. Note that this should only be called by
      * the configuration parser or tests. It is public only for that purpose, mutating enabled state
      * is probably a bad idea.
-     * @param enabled
+     * @param enabled value to set
      */
     public void setEnabled(boolean enabled) {
```

Comment:
I am not sure why the test doesn't seem to pick this up, but since the field is no longer public the test you added is failing.

Reply:
Digging into this, snakeyaml finds `isEnabled` when deciding how to read/write to EncryptionOptions and needs the isEnabled/isOptional types to match exactly.